crates/mongodb-connector/src/api_type_conversions/query_request.rs

```diff
@@ -41,7 +93,13 @@ pub fn v3_to_v2_query_request(
     })
 }
 
-fn v3_to_v2_query(context: &QueryContext, query: v3::Query) -> Result<v2::Query, ConversionError> {
+fn v3_to_v2_query(
+    context: &QueryContext,
+    collection_relationships: &BTreeMap<String, v3::Relationship>,
+    root_collection_object_type: &WithNameRef<schema::ObjectType>,
+    query: v3::Query,
+    collection_object_type: &WithNameRef<schema::ObjectType>,
```

Comment:
I thought about it but if we did that we'd need to stop the caller from being the one constructing and passing the `QueryContext` and make it internal use only. Right now, this struct is used for the caller to pass contextual information, not as a way for this module to have convenient access to "global" information.

I'm going to defer doing this, because frankly this whole module will be deleted when we eventually get rid of the v2 compat layer and go v3 native.